api/v1alpha1/kogitoserverlessworkflow_types.go

```diff
@@ -331,41 +343,57 @@ type Endpoint struct {
 	Protocol string `json:"protocol,omitempty"` // "TCP" or "UDP"; never empty
 }
 
-type StatusCondition string
+// KogitoServerlessWorkflowStatus defines the observed state of KogitoServerlessWorkflowApp
+type KogitoServerlessWorkflowStatus struct {
+	Conditions  BuildStatusCondition         `json:"conditions,omitempty"`
+	Endpoints   []Endpoint                   `json:"endpoints"`
+	Address     duckv1.Addressable           `json:"address,omitempty"`
+	Deployments olm.DeploymentStatus         `json:"deployments"`
+	Phase       ConditionType                `json:"phase,omitempty"`
+	Applied     KogitoServerlessWorkflowSpec `json:"applied,omitempty"`
+	Version     string                       `json:"version,omitempty"`
+}
+
+// ConditionType - type of condition
+type ConditionType string
 
 const (
-	BuildingStatusCondition StatusCondition = "Building"
-	ReadyStatusCondition    StatusCondition = "Ready"
+	// DeployedConditionType - the kieapp is deployed
+	DeployedConditionType ConditionType = "Deployed"
+	// ProvisioningConditionType - the kieapp is being provisioned
+	ProvisioningConditionType ConditionType = "Provisioning"
+	// FailedConditionType - the kieapp is in a failed state
+	FailedConditionType ConditionType = "Failed"
 )
 
-// KogitoServerlessWorkflowStatus defines the observed state of KogitoServerlessWorkflow
-type KogitoServerlessWorkflowStatus struct {
-	Conditions StatusCondition    `json:"conditions,omitempty"`
-	Endpoints  []Endpoint         `json:"endpoints"`
-	Address    duckv1.Addressable `json:"address,omitempty"`
-}
+type BuildStatusCondition string
+
+const (
+	BuildingStatusCondition BuildStatusCondition = "Building"
+	ReadyStatusCondition    BuildStatusCondition = "Ready"
+)
 
-// KogitoServerlessWorkflow is the Schema for the kogitoserverlessworkflows API
+// KogitoServerlessWorkflowApp is the Schema for the kogitoserverlessworkflows API
```

Comment:
Why is this renaming needed?

Reply:
I agree. `KogitoServerlessWorkflow` is the perfect and preferred naming.

Reply:
ok